### PR TITLE
Add tests to Datepicker components

### PR DESCRIPTION
**Pull request by taras**
## Purpose

The goal of this PR is to increase confidence in the Datepicker/Calendar/DayButton components by writing tests for them. 

## Approach

I gathered required requirements about this component from @JohnC-80 and @cherewaty. 

**Features that Datepicker should support**
- uses a standard text input - value can be typed as well as picked using the calendar
- picker disappears with a click outside or blur of the input
- presents in localized format (based on locale to determine input and output)
- custom format can be applied
- excluded dates
- a11y - screen reader compatibility (announces actions, selections, values)
- a11y - keyboard navigation of dates
- accepts colors from a stripes theme/works with stripes-components postcss workflow/css-modules
- supports some version of the “passThroughValue” to keep repetitious entries to a minimum. Use-case: checkin - applying today’s date to multiple items within a check-in.  (would have to discuss alternatives with UX design staff)
- validation styles/classes in-line with other stripes-components
- Optional: outputs date in iso format (output value) (redux: format for input, normalize for output) (assumption is that all folio apps use full ISO, eholdings does not because of ebsco)
 
Then I created a list of all props that this component accepts and went through and ensured that all props either have tests or are marked as deprecated. Here is a list of all props that have been covered by the tests.

**Commonly used props**
- [x] intl
- [x] backendDateStandard
- [x] timeZone
- [x] dateFormat
- [x] locale

**HTML Attributes**
- [x] value
- [x] disabled
- [x] label
- [x] id
- [x] required
- [x] onChange
- [x] readOnly

**Deprecated**
- [x] excludeDates
- [x] ignoreLocalOffset

**Not Deprecated in this PR (but probably should)**
- [x] passThroughValue - when value matches specified value, just pass the value through

## Learning

- [x] Create an issue of accessibility issues in this component - [STCOM-315](https://issues.folio.org/browse/STCOM-315)
- [x] `prop-types-extra` package has [deprecated](https://www.npmjs.com/package/prop-types-extra#deprecatedvalidator-reason) prop type.
- [x] we might want to consider using [react-dates](http://airbnb.io/react-dates/?selectedKind=DateRangePicker%20%28DRP%29&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) instead of fixing this component or writing something from scratch using [downshift](https://github.com/paypal/downshift)

## Screenshots

screen shot 2018-07-18 at 9 19 59 pm

### CODE REVIEW

File: lib/Datepicker/tests/Datepicker-test.js [hunk 1]
```diff
@@ -96,76 +129,204 @@ describe('Datepicker', () => {
...
-  describe('coupled to redux form', () => {
-    describe('selecting a date', () => {
-      let dateOutput;
+  describe('passThroughValue', () => {
+    let onBlur;
+    let onChange;
 
+    beforeEach(async () => {
+      onBlur = mock.fn();
+      onChange = mock.fn();
```
Comment (cherewaty) [inline lib/Datepicker/tests/Datepicker-test.js]: The jest mocks feel a little unit-test-y vs. BigTest-y, but their use makes sense.

File: lib/TextField/stories/BasicUsage.js [hunk 2]
```diff
@@ -5,14 +5,22 @@
...
-      required
     />
+    <form>
+      <TextField
+        label="Required field"
+        placeholder="Placeholder Text"
+        required
+      />
+      <Button type="submit">Submit</Button>
+    </form>
```
Comment (cherewaty) [inline lib/TextField/stories/BasicUsage.js]: What's the additional `<form>` and `<Button>` illustrating on the `<TextField>` story?

File: lib/Datepicker/Datepicker.js [hunk 3]
```diff
@@ -317,20 +321,24 @@ class Datepicker extends React.Component {
...
   }
 
-  handleSetDate(e, date, stringDate) {
+  handleSetDate = (e, date, stringDate) => {
     if (e.type === 'click' || e.type === 'keydown') {
       e.preventDefault();
 
       if (this.props.onChange) { this.props.onChange(e); }
-      this.textfield.current.getInput().focus();
+      // this.textfield.current.getInput().focus();
```
Comment (cherewaty) [inline lib/Datepicker/Datepicker.js]: Leave as-is, delete, or uncomment?

File: lib/Datepicker/tests/Datepicker-Redux-test.js [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+    let onChange;
+
+    beforeEach(async () => {
+      onChange = mock.fn();
+
+      await mountWithContext(
+        <Datepicker
+          input={{ onChange }}
+          meta={{}}
+        />
```
Comment (cherewaty) [inline lib/Datepicker/tests/Datepicker-Redux-test.js]: I've preferred testing `redux-form` like `<Field component={Datepicker} onChange={onChange} />`.

File: lib/TextField/stories/BasicUsage.js (hunk 2 above)
Comment (taras) [reply lib/TextField/stories/BasicUsage.js]: It was to test the required functionality. I wanted to see what it takes to get a validation error. It's essentially submit that causes the validation error to be displayed when an input is required.

File: lib/Datepicker/Datepicker.js (hunk 3 above)
Comment (taras) [reply lib/Datepicker/Datepicker.js]: I'm going to to it back in. It's a left over from accessibility work that I started but it'll need to happen in a separate PR.

File: lib/Datepicker/tests/Datepicker-Redux-test.js (hunk 4 above)
Comment (taras) [reply lib/Datepicker/tests/Datepicker-Redux-test.js]: Ok, I applied `<Field>` to all redux tests

File: lib/Datepicker/Datepicker.js (hunk 3 above)
Comment (taras) [reply lib/Datepicker/Datepicker.js]: Removed

**Pull request merged**